### PR TITLE
Discovery page UI and Integration

### PR DESCRIPTION
**Pull request by HAZberi**
### What this PR does (required):
- Create Discovery Page Minimalist UI and Active contests List and winners list.
- Fetch all contests and filter active contest and integrate data with UI.

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/72773994/127764098-ab9427d3-25fd-4cf9-a1c4-7376ce1de5a5.mov

### Any information needed to test this feature (required):
- Run the app and navigate or click on "Discover" from the navbar to access Discovery Page.

### Any issues with the current functionality (optional):
- Winners list is just front end. Once winners functionality is complete we could easily integrate with UI.
- Social Media Icons are just there to give it a realistic look.

### CODE REVIEW

**Comment by dhanzy** [comment]
LGTM. I thought rajiv said the discovery page should be the home page. Like when user visits the site they should be taken to the discovery page which will be at 
```
http://localhost:3000
```
and not 
```
http://localhost:3000/discovery
```
The only difference between the logged i  user viewing the discovery page is the navbar. 
Ask rajiv to clarify that

**Comment by HAZberi** [comment]
This PR does not specifically deal with routing. Right now discover page is not a protected route and can be accessed without logging in. Once we are finished with all the crucial features, we can create a separate  pull request to setup the routes. Since it will require some changes in the navbar component.

**Comment by dhanzy** [comment]
@HAZberi okay i get.

**Review by dhanzy — APPROVED** [review]

File: client/src/context/useContestContext.tsx [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+  const [allContestsByUser, setAllContestsByUser] = useState<Contest[] | []>([]);
+  const [allContests, setAllContests] = useState<Contest[] | []>([]);
   const { loggedInUser } = useAuth();
   // get all contests and submission by userID
   useEffect(() => {
     if (loggedInUser) {
       const getAllContestByUserId = async () => {
         const contests = await fetchAllContestByUserId({ id: loggedInUser.id });
-        setAllContext(contests.contests);
+        setAllContestsByUser(contests?.contests || []);
```
Comment (rajivtitus) [inline client/src/context/useContestContext.tsx]: Maybe perform a check here (use an if clause) on the response here and accordingly update the state? :)

File: client/src/context/useContestContext.tsx [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
       getAllContestByUserId();
     }
   }, [loggedInUser]);
-  const [allContests, setAllContext] = useState<Contest[]>();
-  return <ContestContext.Provider value={{ allContests: allContests || [] }}>{children}</ContestContext.Provider>;
+
+  useEffect(() => {
+    const getAllContests = async () => {
+      const contests = await fetchAllContests();
+      setAllContests(contests?.contests || []);
```
Comment (rajivtitus) [inline client/src/context/useContestContext.tsx]: Same as above

File: client/src/pages/Discover/ContestCard/ContestCard.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  id: string;
+}
+
+export default function ContestCard({ title, description, prizeAmount, image, deadlineDate, id }: Props): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <Card elevation={2}>
+      <CardActionArea component={Link} to={'/contest/' + id}>
+        <CardMedia>
```
Comment (rajivtitus) [inline client/src/pages/Discover/ContestCard/ContestCard.tsx]: The Card Media component takes an image as a prop, maybe consider using it?

Ref: https://material-ui.com/components/cards/

File: client/src/pages/Discover/WinnersCard/Winner.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      <Grid container>
+        <Grid item xs={12}>
+          <Typography variant="body1">{contestTitle}</Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Typography variant="body1">{moment(winningDate).format('MMM Do, YYYY')}</Typography>
+        </Grid>
+      </Grid>
+      <CardActionArea>
+        <CardMedia>
```
Comment (rajivtitus) [inline client/src/pages/Discover/WinnersCard/Winner.tsx]: Same as comment above for Card Media component

**Review by rajivtitus — APPROVED** [review]
Good work with this! Just some small comments, please address before merging :)

File: client/src/pages/Discover/Discover.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  useEffect(() => {
+    const newActiveContests = allContests.filter((contest: Contest) => new Date(contest.deadlineDate) >= new Date());
+    setActiveContests(newActiveContests);
+  }, [allContests]);
+
+  return (
+    <Grid container className={classes.root} direction="column" alignItems="center">
+      <CssBaseline />
+      <Grid item container>
+        <Navbar />
```
Comment (rajivtitus) [inline client/src/pages/Discover/Discover.tsx:33]: Also, discuss with the team and maybe render the Navbar just once as opposed to rendering it within every component :)

Comment (HAZberi) [reply client/src/pages/Discover/Discover.tsx:33]: I will communicate this with Daniel and tackle the Navbar issue and routing in a separate pull request. Thank you.

File: client/src/context/useContestContext.tsx (hunk 1 above)
Comment (HAZberi) [reply client/src/context/useContestContext.tsx]: Successfully implemented if clause. See implementation at
commit 03e97656998da9be9788a9e83a0d734c45c19266

File: client/src/context/useContestContext.tsx (hunk 2 above)
Comment (HAZberi) [reply client/src/context/useContestContext.tsx]: Successfully implemented at commit 03e97656998da9be9788a9e83a0d734c45c19266

File: client/src/pages/Discover/ContestCard/ContestCard.tsx (hunk 3 above)
Comment (HAZberi) [reply client/src/pages/Discover/ContestCard/ContestCard.tsx]: Successfully implemented at commit 03e97656998da9be9788a9e83a0d734c45c19266

File: client/src/pages/Discover/WinnersCard/Winner.tsx (hunk 4 above)
Comment (HAZberi) [reply client/src/pages/Discover/WinnersCard/Winner.tsx]: Successfully implemented at commit 03e97656998da9be9788a9e83a0d734c45c19266

**Comment by HAZberi** [comment]
Successfully implemented requested changes.

**Pull request merged**